net-vpn/i2p/i2p-2.1.0.ebuild

```diff
@@ -0,0 +1,137 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit java-pkg-2 java-ant-2 systemd
+
+DESCRIPTION="A privacy-centric, anonymous network"
+HOMEPAGE="https://geti2p.net"
+SRC_URI="https://files.i2p-projekt.de/${PV}/i2psource_${PV}.tar.bz2"
+
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="nls test"
+RESTRICT="!test? ( test )"
+
+CP_DEPEND="dev-java/java-service-wrapper:0"
+
+DEPEND="${CP_DEPEND}
+	|| (
+		virtual/jdk:11
+		virtual/jdk:17
+	)
+	nls? ( >=sys-devel/gettext-0.19 )
+	test? (
+		dev-java/ant-junit4:0
+		dev-java/hamcrest-core:1.3
+		dev-java/hamcrest-library:1.3
+		dev-java/junit:4
+		dev-java/mockito:4
+	)
+"
+
+RDEPEND="${CP_DEPEND}
+	acct-user/i2p
+	acct-group/i2p
+	net-libs/nativebiginteger:0
+	|| (
+		virtual/jre:11
+		virtual/jre:17
+	)
+"
+
+EANT_BUILD_TARGET="tarball"
+# no scala as depending on antlib.xml not installed by dev-lang/scala
+EANT_TEST_TARGET="junit.test"
+JAVA_ANT_ENCODING="UTF-8"
+
+src_prepare() {
+	# as early as possible to allow generic patches to be applied
+	default
+
+	java-ant_rewrite-classpath
+
+	java-pkg-2_src_prepare
+
+	# We're on GNU/Linux, we don't need .exe files
+	echo "noExe=true" > override.properties || die
+	if ! use nls; then
+		echo "require.gettext=false" >> override.properties || die
+	fi
+
+	# avoid auto starting browser
+	sed -i "s|clientApp.4.startOnLoad=true|clientApp.4.startOnLoad=false|" \
+		installer/resources/clients.config || die
+
+	# generate wrapper classpath, keeping the default to be replaced later
+	local classpath="wrapper.java.classpath.1=${EPREFIX}/usr/share/i2p/lib/*\n"
+	classpath+="wrapper.java.classpath.2=$(java-pkg_getjars java-service-wrapper)\n"
+
+	# add generated classpath, hardcode system VM, setting system's conf
+	sed -e "s|\(wrapper\.java\.classpath\.1\)=.*|${classpath}|" \
+		-e "s|\(wrapper\.java\.command\)=.*|\1=${EPREFIX}/etc/java-config-2/current-system-vm/bin/java|" \
+		-e "s|\(wrapper\.java\.library\.path\.1\)=.*|\1=${EPREFIX}/usr/$(get_libdir)/java-service-wrapper|" \
+		-e "s|\(wrapper\.java\.library\.path\.2\)=.*|\1=${EPREFIX}/usr/$(get_libdir)/nativebiginteger|" \
+		-e "s|\(wrapper\.java\.additional\.1=-DloggerFilenameOverride\)=.*|\1=${EPREFIX}/var/log/i2p/router-@|" \
+		-e "s|\(wrapper\.logfile\)=.*|\1=${EPREFIX}/var/log/i2p/wrapper|" \
+		-e "/wrapper\.java\.additional\.2\(\.stripquote\|\)/d" \
+		-i installer/resources/wrapper.config ||
```

Comment:
Have you tried getting rid of `installer` (obsoleting `dev-java/java-service-wrapper`)? Gentoo has `ebuild` with `src_install()` ...